math/wxMaxima/files/patch-src_Dirstructure.cpp.diff

```diff
@@ -0,0 +1,37 @@
+--- src/Dirstructure.cpp.orig	2018-02-17 23:54:32.000000000 +0900
++++ src/Dirstructure.cpp	2018-05-31 23:28:55.000000000 +0900
+@@ -249,7 +249,20 @@
+   else
+   {
+     wxFileName maximaName(result);
+-    maximaName.RemoveLastDir();
++    if(maximaName.GetDirCount() > 1)
++    {
++      maximaName.RemoveLastDir();
++    }
++    else
++    {
++      wxArrayString output;
++      wxExecute(wxT("which maxima"), output, (wxEXEC_SYNC | wxEXEC_NOEVENTS));
++      if(output.GetCount() > 0)
++      {
++        wxFileName fullpath = wxFileName::wxFileName(output[0] ,wxPATH_NATIVE);
++        return fullpath.GetPath();
++      }
++    }
+     return maximaName.GetPath();
+   }
+ }
+@@ -279,10 +292,8 @@
+     command = wxT("/Applications/Maxima.app");
+   if (wxFileExists("/Applications/maxima.app"))
+     command = wxT("/Applications/maxima.app");
+-  else if (wxFileExists("/usr/local/bin/maxima"))
+-    command = wxT("/usr/local/bin/maxima");
+-  else if (wxFileExists("/usr/bin/maxima"))
+-    command = wxT("/usr/bin/maxima");
++  else if (wxFileExists("/OPT/LOCAL/bin/maxima"))
++    command = wxT("/OPT/LOCAL/bin/maxima");
```

Comment:
Again, the capitalization here of OPT/LOCAL is somewhat unusual, since even though the file system is case insensitive, this does not correspond to the usual expectations for capitalization.

Reply:
I mean that "/OPT/LOCAL" is a mark of replacement.
After patching, the next command is executed.

    post-patch {
        reinplace -W ${worksrcpath} "s|/OPT/LOCAL|${prefix}|g" \
                                    src/Dirstructure.cpp
    }

Reply:
Most ports use "@@PREFIX@@" as the replacement string, so use that instead of "/OPT/LOCAL".

Reply:
Oh I see. I will change "/OPT/LOCAL" to "@@PREFIX@@".